src/knock.ts

```diff
@@ -75,9 +75,9 @@ class Knock {
    *
    * @param userId {string} The ID of the user that needs a token, e.g. the user viewing an in-app feed.
    * @param options Optionally specify the signing key to use (in PEM or base-64 encoded format), and how long the token should be valid for in seconds
-   * @returns {string} A JWT token that can be used to authenticate requests to the Knock API (e.g. by passing into the <KnockFeedProvider /> component)
+   * @returns {Promise<string>} A JWT token that can be used to authenticate requests to the Knock API (e.g. by passing into the <KnockFeedProvider /> component)
```

Comment:
I don't think it's possible to rewrite it to be synchronous because the underlying web crypto api's are asynchronous: https://github.com/panva/jose/discussions/532

Reply:
Got it okay hm that's unfortunate - we should probably flag/socialize this to the team before cutting a new release.